File: backend/app/schemas/meeting.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel

from app.db.models.meeting import MeetingStatus, MinutesStatus
class MeetingScope(str, Enum):
    HOSTED = "hosted"
    JOINED = "joined"
    SHARED = "shared"
    ALL = "all"


class MeetingHostResponse(BaseModel):
    id: int
    display_name: str


class MeetingParticipantResponse(BaseModel):
    user_id: int | None
    display_name: str
    is_internal: bool


class MeetingPermissionResponse(BaseModel):
    can_view: bool
    can_edit_minutes: bool
    can_regenerate: bool
    can_manage_permissions: bool
    can_view_ai_versions: bool


class MeetingListItemResponse(BaseModel):
    id: int
    subject: str
    meeting_code: str | None
    host: MeetingHostResponse | None
    start_time: datetime | None
    end_time: datetime | None
    meeting_status: MeetingStatus
    ai_minutes_enabled: bool
    minutes_status: MinutesStatus
    my_role: str


class MeetingListDataResponse(BaseModel):
    items: list[MeetingListItemResponse]
    page: int
    page_size: int
    total: int
    total_pages: int


class MeetingDetailResponse(BaseModel):
    id: int
    tencent_meeting_id: str
    meeting_code: str | None
    subject: str
    host: MeetingHostResponse | None
    participants: list[MeetingParticipantResponse]
    start_time: datetime | None
    end_time: datetime | None
    meeting_status: MeetingStatus
    ai_minutes_enabled: bool
    minutes_status: MinutesStatus
    permissions: MeetingPermissionResponse


class ApiSuccessMeetingListResponse(BaseModel):
    code: int = 0
    message: str = "ok"
    data: MeetingListDataResponse
    request_id: str


class ApiSuccessMeetingDetailResponse(BaseModel):
    code: int = 0
    message: str = "ok"
    data: MeetingDetailResponse
    request_id: str

File: frontend/src/types/meeting.ts
export type MeetingScope = 'hosted' | 'joined' | 'shared' | 'all'

export type MeetingStatus = 'SCHEDULED' | 'IN_PROGRESS' | 'ENDED'

export type MinutesStatus =
  | 'DISABLED'
  | 'WAITING_MEETING_END'
  | 'WAITING_RECORDING'
  | 'WAITING_TRANSCRIPT'
  | 'AI_PROCESSING'
  | 'READY'
  | 'FAILED'

export type MeetingRelation = 'HOST' | 'PARTICIPANT' | 'SHARED' | 'ADMIN'

export interface MeetingHost {
  id: number
  display_name: string
}

export interface MeetingParticipant {
  user_id: number | null
  display_name: string
  is_internal: boolean
}

export interface MeetingPermissions {
  can_view: boolean
  can_edit_minutes: boolean
  can_regenerate: boolean
  can_manage_permissions: boolean
  can_view_ai_versions: boolean
}

export interface MeetingListItem {
  id: number
  subject: string
  meeting_code: string | null
  host: MeetingHost | null
  start_time: string | null
  end_time: string | null
  meeting_status: MeetingStatus
  ai_minutes_enabled: boolean
  minutes_status: MinutesStatus
  my_role: MeetingRelation
}

export interface MeetingListData {
  items: MeetingListItem[]
  page: number
  page_size: number
  total: number
  total_pages: number
}

export interface MeetingDetail {
  id: number
  tencent_meeting_id: string
  meeting_code: string | null
  subject: string
  host: MeetingHost | null
  participants: MeetingParticipant[]
  start_time: string | null
  end_time: string | null
  meeting_status: MeetingStatus
  ai_minutes_enabled: boolean
  minutes_status: MinutesStatus
  permissions: MeetingPermissions
}

export interface MeetingListParams {
  scope?: MeetingScope
  meeting_status?: MeetingStatus
  minutes_status?: MinutesStatus
  keyword?: string
  page?: number
  page_size?: number
}
